src/bm_sim/switch.cpp

```diff
@@ -75,14 +75,12 @@ SwitchWContexts::receive(port_t port_num, const char *buffer, int len) {
 
 void
 SwitchWContexts::start_and_return() {
-  {
-    std::unique_lock<std::mutex> config_lock(config_mutex);
-    if (!config_loaded && !enable_swap) {
-      Logger::get()->error(
-          "The switch was started with no P4 and config swap is disabled");
-    }
-    config_loaded_cv.wait(config_lock, [this]() { return config_loaded; });
+  std::unique_lock<std::mutex> config_lock(config_mutex);
+  if (!config_loaded && !enable_swap) {
+    Logger::get()->error(
+        "The switch was started with no P4 and config swap is disabled");
   }
+  config_loaded_cv.wait(config_lock, [this]() { return config_loaded; });
   start();  // DevMgr::start
   start_and_return_();
```

Comment:
A comment here explaining why we are still holding a lock on the `config_mutex` would be nice for future reference.

Reply:
Comment would definitely be helpful. Added!